# Patch to skell-1.13 for the mpc8349itx platform.  This patch is GPL.
# The patch adds a cleaned-up/modified version of the original
# /etc/rc.d/init.d/S31dhcpd script that is used to start/stop/restart
# the DHCP server.  The original file is part of the dhcp package from
# http://www.isc.org/sw/dhcp.
# The patch was originally added for the MPC8349E-mITX platform.
# by: Joe.D'abbraccio@freescale.com on June 1, 2006

diff --exclude CVS -uNr skell-1.13/etc/rc.d/init.d/dhcp skell-1.13.modified/etc/rc.d/init.d/dhcp
--- skell-1.13/etc/rc.d/init.d/dhcp	1969-12-31 19:00:00.000000000 -0500
+++ skell-1.13.modified/etc/rc.d/init.d/dhcp	2006-06-01 11:26:39.000000000 -0400
@@ -0,0 +1,57 @@
+#! /bin/sh
+
+# This file started as /etc/rc.d/init.d/S31dhcpd - Start/Stop the DHCP server
+# daemon(s).
+#
+
+# Comment out the following exit line to enable this script.
+# Before doing so, you need to edit the /etc/dhcpd.conf file.
+#exit 0
+
+KILLWAIT=3
+DAEMON="dhcpd"
+
+which ip
+if [ "$?" != "0" ]; then
+	echo "$0: Missing ip"
+	exit 1
+fi
+
+case "$1" in
+
+    start)
+	# Ensure $DHCP_ARG device is there
+	if ! ip link | grep -q $DHCP_ARG; then
+		echo "$0: Network interface $DHCP not present" >&2
+		exit 2
+	fi
+
+	echo "Starting ${DAEMON}"
+	${DAEMON} $DHCP_ARG
+	if [ "$?" = "0" ]; then echo "Done"
+	else                    echo "FAILED"; fi
+	sleep 1
+	;;
+
+    stop)
+	echo "Stopping ${DAEMON}"
+	killall ${DAEMON}
+	if [ "$?" = "0" ]; then echo "Done"
+	else                    echo "FAILED"; fi
+	;;
+
+    restart)
+	$0 stop
+	sleep 1
+	$0 start
+	;;
+
+    *)
+	echo "Usage: $0 (start|stop|restart)"
+	exit 1
+	;;
+
+esac
+
+exit 0
+
